src/main/java/org/truffleruby/core/module/ModuleNodes.java

```diff
@@ -800,7 +801,8 @@ protected Object classExec(RubyModule self, Object[] args, RubyProc block) {
                     new FixedDefaultDefinee(self),
                     block.declarationContext.getRefinements());
 
-            return callBlockNode.executeCallBlock(declarationContext, block, self, block.block, args);
+            return callBlockNode.executeCallBlock(declarationContext, block, self, block.block,
+                    EmptyArgumentsDescriptor.INSTANCE, args);
```

Comment:
Since this method delegates the arguments to the block it should pass the existing descriptor, isn't it?